### PR TITLE
Replace jQuery tipsy with bootstrap tooltip

### PR DESCRIPTION
**Pull request by Henni**
As requested in #13646 this pull request replaces the jQuery tipsy with the bootstrap tooltip.
I also added a shim for the old tipsy for backwards compatibility.

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
Thanks a lot @Henni 

Please adjust as follows:
- [ ] no minified js libs please ( core/js/tooltip.js )
- [ ] please use bower to "suck in" the libs if possible

**Comment by Henni** [comment]
@DeepDiver1975 fixed the first point but i have no idea how to add this lib to bower. It looks like it is only possible to add the complete bootstrap library.

**Comment by DeepDiver1975** [comment]
> It looks like it is only possible to add the complete bootstrap library.

@MorrisJobke can I ask you to please have a look? THX

**Comment by MorrisJobke** [comment]
> > It looks like it is only possible to add the complete bootstrap library.
> 
> @MorrisJobke can I ask you to please have a look? THX

Naive approach: require bootstrap and add all unwanted files to .gitignore inside the vendor dir ;) (there is already a gitignore with stuff like tests or docs that we already strip out of the packages)

**Comment by Henni** [comment]
Now the import via bower should work.
But I don't know why the last two inspections failed....

@DeepDiver1975 if you wan't me to rebase this just tell me

**Comment by MorrisJobke** [comment]
> But I don't know why the last two inspections failed....

The current CI is heavy overloaded ... hopefully this will be solved in a few hours.

**Comment by jancborchardt** [comment]
Great stuff @Henni, thank you for taking care of this! Also cc @Raydiation for review.

It would be relevant to check if issues https://github.com/owncloud/core/issues/14094 (Tipsy does not close on shared folder) and https://github.com/owncloud/core/issues/15984 (Tooltips do not disappear sometimes using Safari 8.0.5 browser) are also resolved by this.

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

File: core/js/js.js [hunk 1]
```diff
@@ -1742,3 +1742,61 @@ function getScrollBarWidth() {
...
+	if(typeof argument === 'object' && argument !== null) {
+
+		// tipsy defaults
+		var options = {
+			placement: 'bottom',
+			delay: { 'show': 0, 'hide': 0},
+			trigger: 'hover',
+			html: false
+		};
+		console.log(argument);
```
Comment (PVince81) [inline core/js/js.js]: Remove this debugging statement ?

Comment (Henni) [inline core/js/js.js]: @PVince81 Thanks for spotting this. I'll remove it.

File: lib/base.php [hunk 2]
```diff
@@ -392,7 +392,7 @@ public static function initTemplateEngine() {
 		}
 
 		OC_Util::addScript("placeholders");
-		OC_Util::addScript("jquery-tipsy");
+		//OC_Util::addScript("tooltip");
```
Comment (MorrisJobke) [inline lib/base.php]: Can this be removed?

**Comment by MorrisJobke** [comment]
@Henni The tooltips doesn't work anymore. The deprecation notice is shown, but the tooltips are simply not there. For example the tooltip of the upload button that shows the max upload. Can you have a look at this?

**Comment by Henni** [comment]
@MorrisJobke I'll take a look

**Comment by Henni** [comment]
The functionality behind the max upload tipsy writes to the "tipsy-internal" attribute `original-title`.
Bootstrap tooltips use the attribute `data-original-title` instead.
The best solution would be to change the title directly, but bootstrap doesn't detect changes to the title attribute as this issue (twbs/bootstrap#14769) isn't fixed yet.

**Comment by MorrisJobke** [comment]
Works now :) I can't find a not working example :+1:

**Comment by jancborchardt** [comment]
@Henni strange – all the tooltips are defined in the `title` attribute. The `original-title` only is inserted live by Tipsy. So should work.

**Comment by Henni** [comment]
@jancborchardt if you change the title after you initialized the tipsy/tooltip bootstrap and the old tipsy don't detect this. This is why the files app modified `original-title` directly.

The 'current' version of tipsy looks for changes on hover (see http://onehackoranother.com/projects/jquery/tipsy/#notes) but bootstrap-tooltip doesn't (bug!?).

**Comment by jancborchardt** [comment]
@Henni what’s the state of this btw? Ready to review again or does something else need to be fixed/changed?

**Comment by Henni** [comment]
@jancborchardt this is ready to review.

**Comment by jancborchardt** [comment]
@Raydiation can you give this a test since you opened https://github.com/owncloud/core/issues/13646 ?

**Comment by Raydiation** [comment]
:+1:

**Comment by LukasReschke** [comment]
This broke app level tooltips: https://github.com/owncloud/core/issues/17870

**Comment by davitol** [comment]
:+1: Tooltips now work using Safari as browser

**Comment by MorrisJobke** [comment]
> :+1: Tooltips now work using Safari as browser

Nice :)

Thanks @Henni

**Comment by nickvergessen** [comment]
> This broke app level tooltips: #17870

Also broke the headlines of activities (using css `.tooltip`), because they are now font-size 12, opacity 0, ...
I assume this also affects multiple other apps. What's the right replacement for this? Also please drop an email to the mailing list about this BC break

**Pull request merged**